src/mount/linux/mntent.rs

```diff
@@ -0,0 +1,136 @@
+//! Iterate over mtab/fstab
+
+use crate::{NixPath, Result};
+use libc::{endmntent, getmntent, mntent, setmntent, FILE};
+use std::ffi::{CStr, CString};
+
+#[derive(Debug)]
+/// A wrapper for `libc::mntent`, an iterator for `MountEntry`
+pub struct MountEntries {
+    file: *mut FILE,
+}
+
+impl MountEntries {
+    /// Creates a new `MountEntry iterator, opening the given mtab/fstab
+    ///
+    /// # Arguments
+    /// - `path` - Path to mtab/fstab, e.g. `/etc/mtab`.
+    /// - `mode` - Mode as for `fopen(3)`, e.g. `"r"` or `"a+"`.
+    ///
+    /// # See Also
+    /// [`getmntent`](https://www.man7.org/linux/man-pages/man3/getmntent.3.html)
+    /// [`fopen`](https://www.man7.org/linux/man-pages/man3/fopen.3.html)
+    pub fn new<P: ?Sized + NixPath>(path: &P, mode: String) -> Result<Self> {
+        let mode = CString::new(mode).unwrap();
+        let file = path.with_nix_path(|cstr| unsafe {
+            setmntent(cstr.as_ptr(), mode.as_ptr())
+        })?;
+
+        if file.is_null() {
+            panic!("The setmntent call returned NULL");
```

Comment:
I don't like panic! in general in library code. And this function already returns Result. Why not return a suitable Err here?

Reply:
Thanks, the (glibc) man-page wasn't clear about the return of `setmntent` but it returns the same as `fopen`